### PR TITLE
Move Grpc.Auth nuget from grpc/grpc

### PR DESCRIPTION
**Pull request by jtattermusch**
~~Based on https://github.com/grpc/grpc-dotnet/pull/1651~~ (rebased)

Move Grpc.Auth to the grpc-dotnet repository.

### CODE REVIEW

**Comment by jtattermusch** [comment]
@JamesNK this is ready review (even when approved, I still won't merge just yet, I'd like to synchronize removing stuff from grpc/grpc with adding the corresponding stuff to grpc-dotnet).

**Review by JamesNK — APPROVED** [review]
Looks good. I'd like to double-check after Grpc.Core.Api is merged so I can see the Grpc.Auth changes in isolation.

**Comment by jtattermusch** [comment]
> Looks good. I'd like to double-check after Grpc.Core.Api is merged so I can see the Grpc.Auth changes in isolation.

Rebased. PTAL.

File: src/Grpc.Auth/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+bin
+obj
+*.nupkg
```
Comment (JamesNK) [inline src/Grpc.Auth/.gitignore:3]: Is this necessary? There is a .gitignore in the root folder

(I think this came up for the Grpc.Core.Api review)

File: src/Grpc.Auth/GoogleGrpcCredentials.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            var googleCredential = await GoogleCredential.GetApplicationDefaultAsync().ConfigureAwait(false);
+            return googleCredential.ToChannelCredentials();
+        }
+
+        /// <summary>
+        /// Creates an instance of <see cref="CallCredentials"/> that will use given access token to authenticate
+        /// with a gRPC service.
+        /// </summary>
+        /// <param name="accessToken">OAuth2 access token.</param>
+        /// /// <returns>The <c>MetadataCredentials</c> instance.</returns>
```
Comment (JamesNK) [inline src/Grpc.Auth/GoogleGrpcCredentials.cs:52]:
```suggestion
        /// <returns>The <c>CallCredentials</c> instance.</returns>
```

**Review by JamesNK — COMMENTED** [review]
Minor feedback

**Comment by jtattermusch** [comment]
The interop test failure is because we were grpc-dotnet interop against Grpc.Core, which has recently been removed from the master branch on grpc/grpc (see https://github.com/grpc/grpc/pull/29225).
I will fix as a followup.

File: src/Grpc.Auth/.gitignore (hunk 1 above)
Comment (jtattermusch) [reply src/Grpc.Auth/.gitignore]: Removed.

**Pull request merged**